solana/pyth2wormhole/program/src/attest.rs

```diff
@@ -267,44 +260,11 @@ pub fn attest(ctx: &ExecutionContext, accs: &mut Attest, data: AttestData) -> So
         ProgramError::InvalidAccountData
     })?;
 
-    // Adjust message account size if necessary.
-    // NOTE: We assume that:
-    // - the rent and size values are far away from
-    // i64/u64/isize/usize overflow shenanigans (on the order of
-    // single kilobytes).
-    // - Pyth payload size change == Wormhole message size change (their metadata is constant-size)
-    if accs.wh_message.is_initialized() && accs.wh_message.payload.len() != payload.len() {
-        // NOTE: Payload =/= account size (account size includes
-        // surrounding wormhole data structure, payload is just the
-        // Pyth bytes).
-
-        // This value will be negative if we need to shrink down
-        let old_account_size = accs.wh_message.info().data_len();
-
-        // How much payload size changes
-        let payload_size_diff = payload.len() as isize - old_account_size as isize;
-
-        // How big the overall account data becomes
-        let new_account_size = (old_account_size as isize + payload_size_diff) as usize;
-
-        // Adjust account size
-        accs.wh_message.info().realloc(new_account_size, false)?;
-
-        // Exempt balance for adjusted size
-        let new_msg_account_balance = Rent::default().minimum_balance(new_account_size);
-
-        // How the account balance changes
-        let balance_diff =
-            new_msg_account_balance as i64 - accs.wh_message.info().lamports() as i64;
-
-        // How the diff affects payer balance
-        let new_payer_balance = (accs.payer.info().lamports() as i64 - balance_diff) as u64;
-
-        **accs.wh_message.info().lamports.borrow_mut() = new_msg_account_balance;
-        **accs.payer.info().lamports.borrow_mut() = new_payer_balance;
-
-        trace!("After message size/balance adjustment");
-    }
+    let wh_msg_drv_data = P2WMessageDrvData {
+        message_owner: accs.payer.key.clone(),
+        batch_size: batch_attestation.price_attestations.len() as u16,
```

Comment:
also, ideally the client would query the on-chain program to determine the size of any given batch. the on-chain program can have a function like `get_payload_size(number_of_prices)`. This would let the client seamlessly adjust to any attester program updates.

Reply:
Although this seems good and easy in theory, but considering that this size is solely a hack (until wormhole implements resize) I don't think it is worth the effort. We can contribute to wormhole to do that, it's not a hard thing to do.

Querying the on-chain contract in solana is not very simple and it can make smaller tests harder. It probably requires changing ConfigInfo to also add the byte size of each price and has some headache of updating it each time the sdk updates + requires a more thoughtful upgrade of the config account. If we use p2w-sdk to get it from the lib and implement it there then our upgrades could break the client and we should be very careful there.

Also in general, it is unlikely to change price attestation frequently.

Reply:
I actually realized the `v1` literal also has the same issue with upgardes. So, we better be careful with the upgrades.